test/mocha-fixtures.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const mms = require('mongodb-memory-server');
+
+/*
+ * Default MMS mongodb version is used, unless MONGOMS_VERSION is set (which is set with the matrix in test.yml for CI)
+ */
+
+// a single-instance running for single-instance tests
+let mongoinstance;
+
+// a replset-instance for running repl-set tests
+let mongorreplset;
+
+// decide wheter to start a in-memory or not
+const startMemoryInstance = !process.env.MONGOOSE_TEST_URI;
+const startMemoryReplset = !process.env.MONGOOSE_REPLSET_URI;
+
+module.exports.mochaGlobalSetup = async function mochaGlobalSetup() {
+  let instanceuri;
+  let replseturi;
+
+  process.env.RUNTIME_DOWNLOAD = '1'; // ensure MMS is able to download binaries in this context
+
+  // set some options when running in a CI
+  if (process.env.CI) {
+    process.env.MONGOMS_PREFER_GLOBAL_PATH = '1'; // set MMS to use "~/.cache/mongodb-binaries" even when the path does not yet exist
+  }
+
+  if (startMemoryInstance) { // Config to decided if an mongodb-memory-server instance should be used
+    // it's needed in global space, because we don't want to create a new instance every test-suite
+    mongoinstance = await mms.MongoMemoryServer.create({ instance: { args: ['--setParameter', 'ttlMonitorSleepSecs=1'] } });
+    const uri = mongoinstance.getUri();
+    instanceuri = uri.slice(0, uri.lastIndexOf('/'));
```

Comment:
Why `lastIndexOf('/')`? I think we should keep the db name in the connection string...

Reply:
> Why lastIndexOf('/')? I think we should keep the db name in the connection string...

we can keep it, but it would get overwritten anyway again by `test/common.js`, do you know of a better way to overwrite the dbname while keeping everything else while it is still a string (ie in `test/common.js`)?

Reply:
split by / 
pop last element to nirvana
push dbName 
Join by /

Reply:
>  pop last element to nirvana

isnt that what is currently done?

because the original URI would be something like `mongodb://localhost:9999/dbname?queryoptions` in this case `dbname?queryoptions` would be dropped (what currently is done)
though i guess yes we could slice by `/` and `?` (if available)

Reply:
Yeah. 
Or you do a more complex replace with regex.

I think the complexity with regex is not worth it.

Reply:
removed with efaf89b2be02f540c54dac1795b2fcb595a21999
proper handling added in 48973effae9ccea2f62bfe2c7bde3e1fe52620ad